### DISCUSSION TITLE
Upgrading from 2.5 to 2.7

### DISCUSSION THREAD

**Discussion by wfurney** [elsa-workflows/elsa-core#3046] [Q&A] [answered]
I was wondering if I can upgrade from version 2.5 straight to 2.7 or if it is necessary to go from 2.5->2.6->2.7

**Comment by mohdali** [comment, marked as answer]
If there are any database migrations between the version they are going to be applied gradually. So I think it is safe to upgrade directly from 2.5 to 2.7.

However, sometimes the database migrations might fail if you have large number of workflow instances and the execution times out. So it is better to take backups and clear instnace history and execution logs before upgrading the version.

There was one breaking change in 2.5. But since you are already on 2.5 I think you can upgrade to 2.7 easily.

**Reply by wfurney** [reply]
Thanks for the information! I do not know how to clear instance history or execution logs safely, can you please point me towards some documentation or give me a brief overview?

**Reply by sfmskywalker** [reply]
To delete your workflow instances, take a look at your database and remove all records from the following tables:

- Elsa.Bookmarks
- Elsa.WorkflowExecutionLogRecords
- Elsa.WorkflowInstances

**Important**: create a backup first in case you delete something that you did not want to delete.

**Reply by wfurney** [reply]
Thanks!
